x-pack/platform/plugins/shared/alerting/server/lib/to_raw_alert_instances.ts

```diff
@@ -30,7 +31,21 @@ export function toRawAlertInstances<
 
   for (const id of keys(recoveredAlerts)) {
     const alert = recoveredAlerts[id];
-    rawRecoveredAlerts[id] = alert.toRaw(true);
+    if (shouldOptimizeTaskState) {
+      // return recovered alerts if they are flapping or if the flapping array is not at capacity
```

Comment:
the code doesn't seem to be checking anything about capacity? does there need to be a check to see if the flapping history length meets or exceeds the flapping lookback window?

Reply:
I think the comment is misleading, I will update it. We don't need to track the recovered alert if all the values are false, we really only need to care if there is a `true` value. I don't think we need to add a capacity check

Reply:
Fixed in this commit, a74b20f81e8161a41f1459c9274c841ffd790cc7
Thanks for double checking this!